mocks/mocks.py

```diff
@@ -78,9 +82,12 @@ def scale_light_by_distance(particle_positions,
     # Computing the length value that should be added to the positions of
     # particles in the viewing axis (projection axis) as correction.
     physical_distance = ensure_unit(physical_distance, u.Mpc)
+    config = load_yaml_config(config_location)
```

Comment:
@bazkiaei looks like you'll need to load config file at multiple stages (and in other functions) in the mock generation code. Rather than loading it in each function separately, suggest loading it outside of these functions and instead send around the dictionary of all parameters in **kwargs (see e.g. http://book.pythontips.com/en/latest/args_and_kwargs.html ) to all functions. Look at gunagala or POCS for some examples of how you can send parameters everywhere, without listing them one-by-one.

Reply:
This is also something that a class based structure can help with. POCS is actually an example of exactly this, (almost) all of the classes in POCS inherit from the `PanBase` class, and that base class includes loading the config and storing it in a member variable. Consequently all POCS classes automatically have access to the main config, and there's no need to pass it around, either as a dictionary or in `**kwargs`. See: https://github.com/panoptes/POCS/blob/develop/pocs/base.py

POCS does make good use of `**kwargs` though, mostly for flexibility in inheritance and subclassing.

Reply:
Classes are the neatest solution to this, but before you do that I would do what @lspitler suggests. Load the config once, then pass it as an argument to the functions that need it. I would pass it as a single dictionary argument though, instead of using `**` to expand it into keyword arguments. Doing that make it easier to allow the user to override configuration parameters using keyword arguments, and it will also make the transition to classes where the config is a member variable easier.

Reply:
I created an issue to address this suggestion and will open another PR for it very soon.

Reply:
@bazkiaei you should link the issue if you reference it: #67 .

Also, you can now create an Issue directly from a comment. If you click on the `...` to the top-right of the comment there is a `Reference in New Issue` link that will automatically link back.